website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md

```diff
@@ -0,0 +1,23 @@
+
+Paired-end FASTQ files were first converted to an unmapped BAM using Picard's FastqToSam tool with SORT_ORDER = unsorted. (If a read group unmapped BAM file is used as input for the pipeline, this step is skipped.) Unique molecular identifiers (UMIs) were extracted from the unmapped BAM using fgbio's ExtractUmisFromBam and stored in the RX read tag.
+
+After the extraction of UMIs, the unmapped BAM file was aligned to the GENCODE GRCh38 (hg38) v34 with HLA, ALT, and decoy contigs removed (or GENCODE GRCh37 [hg19] v19) using STAR. The --readFilesType and --readFilesCommand parameters were set to "SAM PE" and "samtools view -h", respectively, to indicate that the input was a BAM file. To specify that the output was an unsorted BAM that included unmapped reads, --outSAMtype was set to "BAM Unsorted" and --outSAMunmapped was set to "Within". A transcriptome-aligned BAM was also output with --quantMode = TranscriptomeSAM. To match [ENCODE bulk RNA-seq data standards](https://www.encodeproject.org/data-standards/rna-seq/long-rnas/), the alignment was performed with parameters --outFilterType = BySJout, --outFilterMultimapNmax = 20, --outFilterMismatchNmax = 999, --alignIntronMin = 20, --alignIntronMax = 1000000, --alignMatesGapMax = 1000000, --alignSJoverhangMin = 8, and --alignSJDBoverhangMin = 1. The fraction of reads required to match the reference was set with --outFilterMatchNminOverLread = 0.33 and the fraction of allowable mismatches to read length was set with --outFilterMismatchNoverLmax = 0.1. Chimeric alignments were included with --chimSegmentMin = 15, where 15 was the minimum length of each segment, and --chimMainSegmentMultNmax = 1 to prevent main chimeric segments from mapping to multiple sites. To output chimeric segments with soft-clipping in the aligned BAM, --chimOutType was set to "WithinBAM SoftClip". A maximum of 20 protruding bases at the ends of alignments was allowed with --alignEndsProtrude = 20 ConcordantPair. 
+
+Following alignment, the read group information from the genome-aligned BAM was copied to the transcriptome-aligned BAM using Samtools. Both BAM files were then sorted by coordinate with Picard's SortSam tool, and index and MD5 digest files were created. Duplicate reads were sorted by UMI with UMI-tools to differentiate between biological and PCR duplicates. To specify the tag where the UMIs were stored, --extract-umi-method was set to "tag" and --umi-tag was set to "RX". Unmapped reads were deduplicated with --unmapped-reads = use. Tagged BAM files were output using the option --output-bam. Picard's SortSam tool was used again to sort the BAM files by queryname and create MD5 digest files. PCR duplicates were marked, and duplicate metrics were calculated using Picard's MarkDuplicates tool with GATK v4.1.9.0. BAM files were sorted with Picard's SortSam tool, and index and MD5 digest files were created.
```

Comment:
I think we can omit the first sentence about read group information—it's an implementation detail that normal users don't need to know about.

Reply:
"Duplicate reads were sorted by UMI…" -> "Using umi-tools group, putative duplicates were further divided into subgroups by UMIs and any sequencing errors in UMI bases were corrected."

"Unmapped reads were deduplicated with --unmapped-reads = use." What this argument turned on the unmapped reads are included in the output file as is, rather than discarded. This is important for us, so if the user chooses to reprocess the data in the future, we have not thrown out any reads.

Reply:
Let's talk about this paragraph later.

Reply:
Updated